test/test_selection.py

```diff
@@ -0,0 +1,158 @@
+    )
+    result = selection.remove_disconnected_short_lines(lines_gdf, threshold)
+    assert len(result) == expected_num_lines
+
+
+@pytest.mark.parametrize(
+    (
+        "polygon_gdf",
+        "point_gdf",
+        "expected_containing_point",
+        "expected_not_containing_point",
+    ),
+    [
+        (
+            gpd.GeoDataFrame(
+                {"id": [0, 1, 2]},
+                geometry=[
+                    Polygon([(0, 0), (1, 0), (1, 1), (0, 1)]),
+                    Polygon([(2, 0), (3, 0), (3, 1), (2, 1)]),
+                    Polygon([(4, 0), (5, 0), (5, 1), (4, 1)]),
+                ],
+                crs="EPSG:3067",
+            ),
+            gpd.GeoDataFrame(
+                {"point_id": [0, 1]},
+                geometry=[Point(2.5, 0.5), Point(4.5, 0.5)],
+                crs="EPSG:3067",
+            ),
+            {1, 2},
+            {0},
+        ),
+        (
+            gpd.GeoDataFrame(
+                {"id": [0, 1]},
+                geometry=[
+                    Polygon([(0, 0), (2, 0), (2, 2), (0, 2)]),
+                    Polygon([(3, 0), (5, 0), (5, 2), (3, 2)]),
+                ],
+                crs="EPSG:3067",
+            ),
+            gpd.GeoDataFrame(
+                {"point_id": []},
+                geometry=[],
+                crs="EPSG:3067",
+            ),
+            set(),
+            {0, 1},
+        ),
+    ],
+    ids=[
+        "two_polygons_contain_points_one_not",
+        "no_polygons_contain_points",
+    ],
+)
+def test_split_polygons_by_point_containment(
+    polygon_gdf: gpd.GeoDataFrame,
+    point_gdf: gpd.GeoDataFrame,
+    expected_containing_point: set[int],
+    expected_not_containing_point: set[int],
+):
+    with_points, without_points = selection.split_polygons_by_point_intersection(
+        polygon_gdf, point_gdf
+    )
+
+    assert set(with_points["id"]) == expected_containing_point
+    assert set(without_points["id"]) == expected_not_containing_point
+
+
+@pytest.mark.parametrize(
+    ("lines_gdf", "polygons_gdf", "expected_indices"),
+    [
+        (
+            gpd.GeoDataFrame(
+                {"id": [0, 1, 2]},
+                geometry=[
+                    LineString([(0, 0), (2, 0)]),
+                    LineString([(0.5, 0.5), (1.5, 1.5)]),
+                    LineString([(3, 3), (4, 4)]),
+                ],
+                crs="EPSG:3067",
+            ),
+            gpd.GeoDataFrame(
+                geometry=[Polygon([(0, 0), (2, 0), (2, 2), (0, 2)])], crs="EPSG:3067"
+            ),
+            {1, 2},
+        ),
+        (
+            gpd.GeoDataFrame(
+                {"id": [0]},
+                geometry=[LineString([(0, 0), (1, 1)])],
+                crs="EPSG:3067",
+            ),
+            gpd.GeoDataFrame(geometry=[], crs="EPSG:3067"),
+            {0},
+        ),
+    ],
+    ids=["one_line_on_edge_two_not_on_edge", "empty_polygon_boundary_none"],
+)
+def test_remove_lines_on_polygon_edges(
```

Comment:
As we spoke on Teams, you could expand these test cases with for example crossing lines and lines with partly on boundary and partly not. Then you could add more than one polygon feature to include union and boundary operations to test cases.